### ISSUE TITLE
Publish on package manager repository - vcpkg

### ISSUE THREAD

**Issue by ASil-Scanway** [cieslarmichal/faker-cxx#381]
I'd like to request to publish the package on `vcpkg` repository to enable seamless integration with projects using this package manager.

**Comment by cieslarmichal** [comment]
Thanks for raising the issue! I will take care of that

**Comment by wsekta** [comment]
I could take this issue

**Comment by mostsignificant** [comment]
I am currently working on this, I need some changes in the CMakeLists.txt (mainly install) and then I could publish a port on vcpkg, which includes a patch for the CMakeLists to use fmt from vcpkg if using the faker-cxx via vcpkg. If interested, I can submit a PR but I need a CMake guru to review it since my CMake understanding is limited

**Comment by cieslarmichal** [comment]
I am not a guru but I will try to review it :)

**Comment by cieslarmichal** [comment]
Hello @mostsignificant are you working on this?

**Comment by mostsignificant** [comment]
@cieslarmichal Yes, please review my PR - some little caveats though, especially on MacOS X

**Comment by cieslarmichal** [comment]
looks good, merged :) what are next steps here?

**Comment by cieslarmichal** [comment]
good job :fire:

**Comment by mostsignificant** [comment]
I will submit a PR for vcpkg tomo, I have already prepared the files (vcpkg.json, portfile.cmake, usage, and two patch files) for a port. If accepted, I will add a section about vcpkg in your README

**Comment by mostsignificant** [comment]
Okay, I already opened the Draft PR, you can check it out here: https://github.com/microsoft/vcpkg/pull/38583

**Comment by cieslarmichal** [comment]
Nice

**Comment by AdamJovanovic** [comment]
Would it make sense for `googletest` to also be a dependency of faker-cxx when using vcpkg?

**Comment by cieslarmichal** [comment]
Good question, I don't think we need that there

**Comment by mostsignificant** [comment]
We could add it as vcpkg port "feature" - which means you could build with tests enabled and it also includes googletest in this case. I am not sure about doing the same for fmt so users must only activate it when doing a Apple clang 12 build

**Comment by mostsignificant** [comment]
Almost finished with the initial version, but I need some help here:
https://github.com/microsoft/vcpkg/pull/38583/files/64afeaec8a9f0f3cddd3b309f25851dfd6c9f9f7#diff-589dc7ff9a891c76d3949346ccbe3848bd4289668fde9f8c0cc4bbd2ce04a3ef

Does this mean that faker-cxx needs an extra <x>Config.cmake file? In this case I would need to create them in faker-cxx first and then the referenced version in the vcpkg ports needs to be updated. But I don't know how to write those config.cmake files, I initially thought the install commands in the CMakeLists.txt file take care of it

**Comment by cieslarmichal** [comment]
Hey, honestly I have no idea how to deal with it. I will ask in our discord channel.

**Comment by AdamJovanovic** [comment]
I'm having a read of this right now [CMake Export Configuration](https://cmake.org/cmake/help/latest/guide/tutorial/Adding%20Export%20Configuration.html)
Still trying to figure it out, but it seems like it's along the lines of what we need to fix. Posting it incase you guys understand it better

**Comment by icyfox168168** [comment]
请继续这工作，vcpkg真很方便

**Comment by cieslarmichal** [comment]
done https://github.com/microsoft/vcpkg/pull/46286
